### PR TITLE
Added code to support Azure Cognitive Servies auth token

### PR DESCRIPTION
**Pull request by devdupont**
I've added support for the new/reimplemented Azure Cognitive Servies API. Thankfully, the API calls are all the same except for how the 10-minute auth token is fetched. Instead of using a client_id and client_key, it now uses a single api_key to fetch the token.

To start, I created a class called AzureAuthToken which stores the auth token value. The object is initialized with the api_key, and the token value is called via my_auth_token.value which is a class property. This allows the class to return a value which is always valid by checking how old the stored value is.

This also allows the AzureAuthToken to be called and maintained separate from any single Translator instance. I have the following code running in a production Flask app:

```python
AUTHTOKEN = AzureAuthToken(AZURE_TRANS_KEY)

def azure_translate(bookmark_id: int, to: str):
    bookmark = Bookmark.query.filter(Bookmark.id == bookmark_id).first()
    text = (bookmark.title or '', bookmark.body or '')
    translator = Translator(None, None)
    translator.access_token = AUTHTOKEN.value
    return [t['TranslatedText'] for t in translator.translate_array(text, to)]
```

I later decided to fork your code to add this support as well. I added the AzureAuthToken class into __init__.py. I also added a second version of the Translator class called TranslatorCS which is identical except for how the auth token is fetched.

- __init__ only takes an api_key and debug and creates an AzureAuthToken instance
- call changed how the token is called
- get_access_token raises an attribute error

The goal here was to change as little as possible while making it as simple to call on the client side. The user calls the exact same code just with a different class name and init variables.

```python
>>> t = TranslatorCS('my_api_key')
>>> t.translate('Hello!', 'es')
'¡Hola!'
```

However, feel free to change it as you see fit. There's a lot of promise by fully integrating the AzureAuthToken class into both versions of Translator. As I show above, it's possible to optimize a backend server by separating out the auth token.

### CODE REVIEW

**Comment by sharoonthomas** [comment]
Closing and re-opening for travis to build

File: __init__.py [hunk 1]
```diff
@@ -227,3 +227,78 @@ def detect_language(self, text):
             'text': text.encode('utf8')
         }
         return self.call('Detect', params)
+
+class AzureAuthToken:
+    """Class to make sure that .value is always a valid 10-min auth token"""
+    _token = None
+    last_fetched = None
+
+    def __init__(self, api_key: str):
```
Comment (sharoonthomas) [inline __init__.py:236]: type hint support is not available on 2.x and < 3.5

**Comment by sharoonthomas** [comment]
@flyinactor91 thanks for the patch.

I know its been a while but would you be able to make a couple of changes?

1. The library should support a lot of python 2.X that could still be using this
2. Instead of another class, can we have an api that look something like

```
from microsofttranslator import Translator
translator = Translator(api_token='<token>')
```

this means making `client_id` and `client_secret` optional and introducing another keyword argument `api_token` and then the class itself managing its auth. Thoughts?

**Review by sharoonthomas — CHANGES_REQUESTED** [review]

File: __init__.py (hunk 1 above)
Comment (newearthmartin) [reply __init__.py:236]: I confirm that by removing the :str it works in python 2

**Comment by newearthmartin** [comment]
Hey guys what's the status of this pull request and the library in general?
What's needed in order to merge this PR? 

According to an email from microsoft, we need also to migrate to V3 of the API before april. It would be nice to use this opportunity to merge this PR and also update to V3.

**Comment by sharoonthomas** [comment]
@newearthmartin happy to merge a PR that will work for both python 2 and 3. The only problem with the current patch is that it will make the library py3 only.

**Comment by newearthmartin** [comment]
@sharoonthomas this is why I created this other PR https://github.com/fulfilio/Microsoft-Translator-Python-API/pull/3 which is essentially the same but removing type hinting. I use it in my python 2 project

**Comment by newearthmartin** [comment]
no longer needed, please close

**Pull request closed without merging**